service/telemetry/config.go

```diff
@@ -137,9 +127,45 @@ type TracesConfig struct {
 func (c *Config) Validate() error {
 
 	// Check when service telemetry metric level is not none, the metrics address should not be empty
-	if c.Metrics.Level != configtelemetry.LevelNone && c.Metrics.Address == "" {
+	if c.Metrics.Level != configtelemetry.LevelNone && c.Metrics.Address == "" && len(c.Metrics.Readers) == 0 {
 		return fmt.Errorf("collector telemetry metric address should exist when metric level is not none")
 	}
 
 	return nil
 }
+
+func (mrs *MeterProviderJsonMetricReaders) Unmarshal(conf *confmap.Conf) error {
+	if err := conf.Unmarshal(mrs); err != nil {
+		return err
+	}
+	if *mrs == nil {
+		*mrs = make(MeterProviderJsonMetricReaders)
+	}
+	for key, reader := range *mrs {
+		readerType := strings.Split(key, "/")[0]
+		switch readerType {
+		case "pull":
+			var r PullMetricReader
+			if err := mapstructure.Decode(reader, &r); err != nil {
+				return fmt.Errorf("invalid pull metric reader configuration: %w", err)
+			}
+			(*mrs)[key] = r
+
+			for key, exporter := range r.Exporter {
+				switch key {
+				case "prometheus":
+					var promExporter Prometheus
+					if err := mapstructure.Decode(exporter, &promExporter); err != nil {
+						return fmt.Errorf("invalid exporter configuration: %w", err)
+					}
+					r.Exporter[key] = promExporter
+				default:
+					return fmt.Errorf("unsupported metric exporter type: %s", key)
```

Comment:
nit (for consistency with other errors such as https://github.com/open-telemetry/opentelemetry-collector/blob/bc9421c14b0e856b36d9ad9545e423bf8112f99f/config/configtelemetry/configtelemetry.go#L76)
```suggestion
					return fmt.Errorf("unsupported metric exporter type %q", key)
```